lambdas/check-coi/src/main/java/uk/gov/di/ipv/core/checkcoi/CheckCoiHandler.java

```diff
@@ -152,6 +152,27 @@ public Map<String, Object> handleRequest(ProcessRequest request, Context context
                                 .areFamilyNameAndDobCorrelatedForCoiCheck(combinedCredentials);
                         case FULL_NAME_AND_DOB -> userIdentityService.areVcsCorrelated(
                                 combinedCredentials);
+                        case ANY_VALID -> {
+                            if (oldVcs.isEmpty()) {
+                                anyValidPassed = "No existing identity";
+                                yield true;
+                            }
+                            if (userIdentityService.areGivenNamesAndDobCorrelated(
+                                    combinedCredentials)) {
+                                anyValidPassed = GIVEN_NAMES_AND_DOB.toString();
+                                yield true;
+                            }
+                            if (userIdentityService.areFamilyNameAndDobCorrelatedForCoiCheck(
+                                    combinedCredentials)) {
+                                anyValidPassed = FAMILY_NAME_AND_DOB.toString();
+                                yield true;
+                            }
+                            if (userIdentityService.areVcsCorrelated(combinedCredentials)) {
+                                anyValidPassed = FULL_NAME_AND_DOB.toString();
+                                yield true;
+                            }
```

Comment:
Is it possible for this to pass, if the two previous checks have failed? If it's an interesting distinction, we should probably run the checks from strictest to weakest.

Reply:
That's a good point - and I think that for what we're doing the full name check isn't necessary. I'm going to wait for @thebauSoftwire's PR to get merged, as I think that will mean there's no need to add the new check type. We can just use the new one created in that PR.